### ISSUE TITLE
Warn/Error out if CMAKE_BUILD_TYPE is unrecognized

### ISSUE THREAD

**Issue by StefanBruens** [vegastrike/Vega-Strike-Engine-Source#138]
The CMakeLists.txt only handles Debug/Release/Profiler correctly, otherwise some defines (e.g. -DNO_PY_SIGNATURES) will be missing and the build fails.

Many distributions use "RelWithDebinfo" as default build type, which leads to somewhat obscure build failures.

**Comment by BenjamenMeyer** [comment]
Thanks; we'll certainly address this.

I'm putting it on the backlog for now only because I think it'll likely have to get done as part of refactoring the code structure and our use of CMake. But it could certainly be done earlier.

**Comment by stephengtuggy** [comment]
@StefanBruens Thank you for reporting this. At least part of the problem has now been addressed. I added the RelWithDebInfo build type back in in #202 , so it is now a valid build configuration once more.

**Comment by BenjamenMeyer** [comment]
@stephengtuggy @StefanBruens is this resolved? Can we close this?

**Comment by stephengtuggy** [comment]
@BenjamenMeyer No, it's not completely resolved. We still need to validate that the value of `CMAKE_BUILD_TYPE` is one of the valid options.

**Comment by BenjamenMeyer** [comment]
@stephengtuggy can we close this? we're at least doing something with:
https://github.com/vegastrike/Vega-Strike-Engine-Source/blob/master/engine/CMakeLists.txt#L92-L95

Though per #56 and #889 we probably should move that logic into the root CMakeLists.txt

**Comment by stephengtuggy** [comment]
@BenjamenMeyer we can probably close this. I think the original issue that @StefanBruens reported is fixed. We could still add a check that the specified CMAKE_BUILD_TYPE is one of the valid options. I think that could be treated as a separate issue though.

**Comment by BenjamenMeyer** [comment]
we can always re-open if it's an issue
